src/integer_mod_q/poly_over_zq/norm.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright © 2023 Phil Milewski
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module includes functionality to compute several norms
+//! defined on polynomials.
+
+use crate::{
+    integer::Z,
+    integer_mod_q::PolyOverZq,
+    traits::{GetCoefficient, Pow},
+};
+
+impl PolyOverZq {
+    /// Returns the squared Euclidean norm or 2-norm of the given polynomial.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::{integer::Z, integer_mod_q::PolyOverZq};
+    /// use std::str::FromStr;
+    ///
+    /// let poly = PolyOverZq::from_str("3  1 2 3 mod 11").unwrap();
+    ///
+    /// let sqrd_2_norm = poly.norm_eucl_sqrd();
+    ///
+    /// // 1*1 + 2*2 + 3*3 = 14
+    /// assert_eq!(Z::from(14), sqrd_2_norm);
+    /// ```
+    pub fn norm_eucl_sqrd(&self) -> Z {
+        let mut res = Z::ZERO;
+        for i in 0..=self.get_degree() {
+            let mut coeff: Z = self.get_coeff(i).unwrap();
+            let minus_coeff = Z::from(&self.modulus) - &coeff;
+            if minus_coeff < coeff {
+                //todo: use min once on dev
+                coeff = minus_coeff;
+            }
+            res = res + coeff.pow(2).unwrap();
```

Comment:
rather use the distance function for Zq values, I think this would improve readability here